code/code/misc/shop.cc

```diff
@@ -1852,21 +1852,21 @@ void shopping_list(sstring argument, TBeing *ch, TMonster *keeper, int shop_nr)
     // check class restriction
     extra_flags = convertTo<int>(db["extra_flags"]);
       
-    fit=true;
-    if(ch->hasClass(CLASS_MAGE) && (extra_flags & ITEM_ANTI_MAGE))
-      fit=false;
-    if(ch->hasClass(CLASS_CLERIC) && (extra_flags & ITEM_ANTI_CLERIC))
-      fit=false;
-    if(ch->hasClass(CLASS_WARRIOR) && (extra_flags & ITEM_ANTI_WARRIOR))
-      fit=false;
-    if(ch->hasClass(CLASS_THIEF) && (extra_flags & ITEM_ANTI_THIEF))
-      fit=false;
-    if(ch->hasClass(CLASS_SHAMAN) && (extra_flags & ITEM_ANTI_SHAMAN))
-      fit=false;
-    if(ch->hasClass(CLASS_DEIKHAN) && (extra_flags & ITEM_ANTI_DEIKHAN))
-      fit=false;
-    if(ch->hasClass(CLASS_MONK) && (extra_flags & ITEM_ANTI_MONK))
-      fit=false;
+    fit=false;
+    if(ch->hasClass(CLASS_MAGE) && !(extra_flags & ITEM_ANTI_MAGE))
+      fit=true;
+    if(ch->hasClass(CLASS_CLERIC) && !(extra_flags & ITEM_ANTI_CLERIC))
+      fit=true;
+    if(ch->hasClass(CLASS_WARRIOR) && !(extra_flags & ITEM_ANTI_WARRIOR))
+      fit=true;
+    if(ch->hasClass(CLASS_THIEF) && !(extra_flags & ITEM_ANTI_THIEF))
+      fit=true;
+    if(ch->hasClass(CLASS_SHAMAN) && !(extra_flags & ITEM_ANTI_SHAMAN))
+      fit=true;
+    if(ch->hasClass(CLASS_DEIKHAN) && !(extra_flags & ITEM_ANTI_DEIKHAN))
+      fit=true;
+    if(ch->hasClass(CLASS_MONK) && !(extra_flags & ITEM_ANTI_MONK))
+      fit=true;
```

Comment:
It'd be more honest if fit = none(anti_bits), instead of !all(anti_bits). It'd be a major nerf for multiclassers, though.
@Aion2501 opinions?

Reply:
Multi-classes could use a kick in the pants for pracs. 
Depending on the class, they're getting an extra 1-3 pracs per level for the first 30 levels. Single classes have to max basic 200 points before their prac cap is lifted. Usually something like 7 pracs then 9 starting around Lvl 30.
(because Peel ripped out the INT factor and added a flat modifier)

Multi-classes don't have that restriction in the first place (!bug) and even if they did, they can max the first 200 much, much faster. So they're dripping with extra pracs. About 90 I'd say, on average.

Reply:
This would make it so Mage/Cleric/Warrior couldn't use a piece of armor that was anti-cleric for example?

HUGE downvote from me on that one if that's the case. We're not equipped to deal with that change, because none (or very few) of the objects in the game were designed with that in mind. Tri-classes would go naked in the streets. 

The alternative idea is to implement the additional separation of objects by type. Heavy Armor, Medium Armor, and Light Armor are sub-types already written in and abandoned. 
Heavy - War/Deik
Medium - Cleric/Ranger
Light - Thief/Mage
Cloth - Monk/Shaman

You'd have to force these flags upon everything in the game based on the AC level the object has, because there are way too many objects for me to do it by hand. Or some other ingenious method. Same reason the anti-flags would be so troublesome. But if you accomplish that, then we can add the armor req to the class kits and force multiclassers to use the lowest kit in their setup.

Reply:
The change here is to which items are listed in a shop when you type "list fit". It in NO way affects what a multiclass can wear. It's a defect that typing "list fit" doesn't currently display items that a character can wear.
--
If we want to change what multiclasses can wear that's a balance discussion that we should start somewhere else.

Reply:
Phew. Ok. Approve than. Don't mind me. You know I can only read about 10% of this code.